streamerate/streams.py

```diff
@@ -590,10 +592,81 @@ def flatMap(self, predicate: Callable[[_K], Iterable[_V]] = _IDENTITY_FUNC) -> "
             return stream(ItrFromFunc(lambda: itertools.chain.from_iterable(self)))
         return stream(ItrFromFunc(lambda: itertools.chain.from_iterable(self.map(predicate))))
 
+    def makeMapping(
+        self: "stream[_K]", f: Callable[[_K], _V]
+    ) -> "stream[Tuple[_K, _V]]":
+        """Apply a unary function to a stream of values x and produce a
+        stream of tuples (x, f(x)). Useful for chaining with `.toMap()`.
+        """
+        return self.map(lambda x: (x, f(x)))
+
+    def makeInverseMapping(
```

Comment:
`makeInverseMapping`: consider using `pairBy`, which is concise and mirrors `pairWith`, indicating that the function is applied to produce the first element (key) in each tuple.